File: server/Domain/Entities/Transaction.cs
namespace Domain.Entities
{
public class Transaction
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public string Description { get; set; }
    public DateTime Date { get; set; }

    public int BudgetId { get; set; }
    public Budget Budget { get; set; }

    public int CategoryId { get; set; }
    public Category Category { get; set; }
}

}

File: client/src/types/Transaction.ts
export interface TransactionResponse {
  id?: number;
  amount: number;
  description?: string;
  date: string;
  budgetId: number;
  budgetName?: string;
  categoryId: number;
  categoryName?: string;
}

export interface CreateTransactionRequest {
  amount: number;
  description?: string;
  categoryId: number;
  budgetId: number;
  date: string;
}

export interface UpdateTransactionRequest {
  id: number;
  amount: number;
  description?: string;
  date: string;
  budgetId: number;
  categoryId: number;
}
